src/data/bucket/line_attributes.js

```diff
@@ -1,10 +1,17 @@
 // @flow
 import { createLayout } from '../../util/struct_array';
 
-const layout = createLayout([
+export const lineLayoutAttributes = createLayout([
     {name: 'a_pos_normal', components: 4, type: 'Int16'},
     {name: 'a_data', components: 4, type: 'Uint8'}
 ], 4);
 
-export default layout;
-export const {members, size, alignment} = layout;
+// these attributes are only needed if using data-driven line-pattern
+export const linePatternAttributes = createLayout([
+    // [tl.x, tl.y, br.x, br.y]
+    {name: 'a_pattern_from', components: 4, type: 'Float32'},
+    {name: 'a_pattern_to', components: 4, type: 'Float32'}
+]);
+
+export default lineLayoutAttributes;
+export const {members, size, alignment} = lineLayoutAttributes;
```

Comment:
Since this module now exports more than one set of attributes, let's drop the default export in favor of two named exports (See https://github.com/mapbox/mapbox-gl-js/issues/6336)